glide-core/src/client/mod.rs

```diff
@@ -525,11 +525,27 @@ impl Client {
                     Some(ResponsePolicy::AllSucceeded),
                 ));
                 let mut cmd = redis::cmd("AUTH");
+                if let Some(username) = self.get_username().await {
+                    cmd.arg(username);
+                }
                 cmd.arg(password);
                 self.send_command(&cmd, Some(routing)).await
             }
         }
     }
+
+    /// Returns the username if one was configured during client creation. Otherwise, returns None.
+    async fn get_username(&mut self) -> Option<String> {
+        match &mut self.internal_client {
+            ClientWrapper::Cluster { client } => match client.get_username().await {
+                Ok(Value::SimpleString(username)) => Some(username),
+                Ok(Value::Nil) => None,
+                Ok(other) => panic!("Expected SimpleString, got: {:?}", other),
+                Err(e) => panic!("Got error trying to get username: {:?}", e),
```

Comment:
We don't want to panic if get_username fails, as it would crash the whole client. Instead, change get_username to return Result<Option<String>> and return the error if we get one.